reacttodo/src/App.js

```diff
@@ -0,0 +1,93 @@
+import "./App.css";
+import React, { useState, useEffect } from "react";
+
+function App() {
+  const [data, setData] = useState([]);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState("");
+
+  useEffect(() => {
+    const getTodoData = JSON.parse(localStorage.getItem("todoData")) ?? [];
+    setData(getTodoData);
+  }, []);
+
+  const submitHandler = () => {
+    if (!title || !description) return;
+    const currentData = [{ title, description }, ...data];
+    setData(currentData);
+    localStorage.setItem("todoData", JSON.stringify(currentData));
+    setDescription("");
+    setTitle("");
+  };
+
+  const deleteItem = (idx) => {
+    const updateData = data.filter((_, id) => idx !== id);
+    setData(updateData);
+    localStorage.setItem("todoData", JSON.stringify(updateData));
+  };
+  const updateItem = (todo, idx) => {
+    setDescription(todo.description);
+    setTitle(todo.title);
+    setData((arr) => {
+      return arr.filter((_, id) => idx !== id);
+    });
+  };
+  const searchItem = (searchValue) => {
+    const data = JSON.parse(localStorage.getItem("todoData")) ?? [];
+    const searchData = data.filter(
+      (obj) =>
+        obj.title.includes(searchValue) || obj.description.includes(searchValue)
+    );
+
+    setData(searchData);
+  };
+  return (
+    <div className="App">
+      <h1>React Note-taking App</h1>
+      <label htmlfor>Search Here &nbsp;</label>
+      <input type="search" onChange={(e) => searchItem(e.target.value)} />
+      <form onSubmit={(e) => e.preventDefault()}>
+        <label htmlfor="title">Title</label>
+        <input
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
+          id="title"
+          type="text"
+        />
+        <label htmlfor="description">Description</label>
+        <textarea
+          value={description}
+          onChange={(e) => setDescription(e.target.value)}
+          id="description"
+        ></textarea>
+        <button onClick={() => submitHandler()} type="submit">
+          Submit
```

Comment:
@isagarjaiswal I can see you are using the same form to update too, right? Can you please also handle this if this form is used for update change the button text to `Update`